packages/location-management/src/components/LocationUnitList/index.tsx

```diff
@@ -147,7 +147,7 @@ export const LocationUnitList: React.FC<Props> = (props: Props) => {
     }
   }, [treeData, currentClicked]);
 
-  if (loading) return <Spin size={'large'} />;
+  if (loading || !tableData.length) return <Spin size={'large'} />;
```

Comment:
Will this stop loading if you receive tableData that is legitimately an empty array?

Reply:
Replaced checking for tableData's length to checking if tableData has been evaluated (irregardless of it's length)